lib/commands/devices/supported.ts

```diff
@@ -69,45 +69,50 @@ export default class DevicesSupportedCmd extends Command {
 		}),
 		verbose: flags.boolean({
 			char: 'v',
-			description:
-				'add extra columns in the tabular output (ALIASES, ARCH, STATE)',
+			description: 'add extra columns in the tabular output',
 		}),
 	};
 
 	public async run() {
 		const { flags: options } = this.parse<FlagsDef, {}>(DevicesSupportedCmd);
-		const dts = await getBalenaSdk().models.config.getDeviceTypes();
-		let deviceTypes: Array<Partial<SDK.DeviceTypeJson.DeviceType>> = dts.map(
-			(d) => {
-				if (d.aliases && d.aliases.length) {
-					// remove aliases that are equal to the slug
-					d.aliases = d.aliases.filter((alias: string) => alias !== d.slug);
-					if (!options.json) {
-						// stringify the aliases array with commas and spaces
-						d.aliases = [d.aliases.join(', ')];
-					}
-				} else {
-					// ensure it is always an array (for the benefit of JSON output)
-					d.aliases = [];
-				}
-				return d;
-			},
-		);
+		const [dts, configDTs] = await Promise.all([
+			getBalenaSdk().models.deviceType.getAll({
+				$expand: { is_of__cpu_architecture: { $select: 'slug' } },
+			}),
+			getBalenaSdk().models.config.getDeviceTypes(),
+		]);
+		const configDTsBySlug = _.keyBy(configDTs, (o) => o.slug);
+		interface DT {
+			slug: string;
+			aliases: string[];
+			arch: string;
+			state: string;
+			name: string;
+		}
+		let deviceTypes: DT[] = [];
+		for (const dt of dts) {
+			const configDT = configDTsBySlug[dt.slug] || {};
+			const aliases = (configDT.aliases || []).filter(
+				(alias) => alias !== dt.slug,
+			);
+			const arch = (dt.is_of__cpu_architecture as any)?.[0]?.slug;
+			deviceTypes.push({
+				slug: dt.slug,
+				aliases: options.json ? aliases : [aliases.join(', ')],
+				arch: arch || 'n/a',
+				// 'BETA' renamed to 'NEW'
+				// https://www.flowdock.com/app/rulemotion/i-cli/threads/1svvyaf8FAZeSdG4dPJc4kHOvJU
+				state: (configDT.state || 'NEW').replace('BETA', 'NEW'),
```

Comment:
I found that `models.deviceType.getAll()` was listing some 11 additional device types compared to `models.config.getDeviceTypes()`, so `configDT.state` above would be `undefined`. I assumed, perhaps incorrectly, that those were new device types, hence `configDT.state || 'NEW'`.